foundry/applications/formApplications/setupConfigurationForm.d.ts

```diff
@@ -1,3 +1,206 @@
-declare class SetupConfigurationForm extends FormApplication {
-  _updateObject(event?: Event, formData?: object): Promise<any>; // TODO: Add correct return type
+/**
+ * The Package Configuration setup application
+ */
+declare class SetupConfigurationForm extends FormApplication<SetupConfigurationForm.Data> {
+  constructor({
+    systems,
+    modules,
+    worlds,
+    currentWorld
+  }: {
+    systems: SetupConfigurationForm['systems'];
+    modules: SetupConfigurationForm['modules'];
+    worlds: SetupConfigurationForm['worlds'];
+    currentWorld?: SetupConfigurationForm['currentWorld'];
+  });
+
+  /**
+   * Valid Game Systems to choose from
+   */
+  systems: Game.System[];
+
+  /**
+   * Install Modules to configure
+   */
+  modules: Game.Module[];
+
+  /**
+   * The Array of available Worlds to load
+   */
+  worlds: Game.World[];
+
+  /**
+   * The currently inspected World
+   * @defaultValue `null`
+   */
+  currentWorld: string | null;
+
+  /**
+   * The currently viewed tab
+   */
+  protected _tab: 'worlds';
+
+  /**
+   * Track the button elements which represent updates for different named packages
+   * @defaultValue `null`
+   */
+  protected _progressButton: HTMLElement | null;
+
+  /**
+   * @override
+   * @defaultValue
+   * ```typescript
+   * mergeObject(super.defaultOptions, {
+   *   id: "setup-configuration",
+   *   classes: ["dark"],
+   *   template: "templates/setup/setup-config.html",
+   *   popOut: false,
+   *   scrollY: ["#world-list", "#system-list", "#module-list"],
+   *   tabs: [{navSelector: ".tabs", contentSelector: ".content", initial: "worlds"}]
+   * });
+   * ```
+   */
+  static get defaultOptions(): FormApplication.Options;
+
+  /**
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): SetupConfigurationForm.Data;
+
+  static tagPackageAvailability(pkg: Game.Module | Game.System | Game.World): void;
+
+  /** @override */
+  activateListeners(html: JQuery): void;
+
+  /**
+   * Post the setup configuration form
+   */
+  protected _post(data: { action: string; world?: string }): Promise<string | { error?: string; stack?: string }>;
```

Comment:
I believe the error case thing is dead code in foundry because all the submit buttons cause a redirect. Also it would probably be better to type the return value as `Record<string, unknown> & {error?: string; stack?: string}` in that case.